### ISSUE TITLE
[pt-br] The word "câmera" is not archaic

### ISSUE THREAD

**Issue by hmslima** [languagetool-org/languagetool#3095]
Language Tool labels the word "câmera" as an archaic word, but that is totally incorrect as you can see in the Wikipedia article from the link below:

https://pt.wikipedia.org/wiki/C%C3%A2mera

In Brazil, we say "câmera fotográfica" while in Portugal it's said "câmara fotográfica".

**Comment by danielnaber** [comment]
@hmslima Thanks, I have removed that word from the list.

@marcoagpinto Feel free to move that word/rule to `pt-PT/grammar.xml` (it was in `pt/grammar.xml`)

**Comment by danielnaber** [comment]
@hmslima Would you be interested in going over all the expressions that are supposed to be archaic to check them? They are listed as regular expressions, which makes it a bit hard to read:

* a(?:bominábil|borrimento|bsentar|bsumir|ceiro|chacar|coiteza|contista|có|dvogacia|guazil|H|lacrão|lantíase|lbora|lfaça|lgibebe|lguazil|lgur|lgures|lmeida|lmenara|lvazir|mictório|morar|mículo|nrique|ntigüidade|ntão|periente|quessa|quesse|questa|queste|questo|rcocentro|tá|utopse|çucre)|b(?:abá|andir|anguê|enção|oa-venturança|odalha|ofé|ródio|uliço)
* Brazil
* c(?:afofo|alendairo|arpinteiragem|assar|ham|hara|oidar|omest[ao]|ommummente|omua|oncassor|onvescote|recimento|relgo|réligo|ã|ôvado)
* d(?:esemparar|esnobre|evisar|iabro|iese|ino|iésis|olor|orosamente|oroso|díesis)
* e(?:mpacar|ngrês|ntonce|ntrelocutor|sto|xprimentava)
* f(?:acha|atexa|enestra|imatíase|imia|logose)
* g(?:alactooligossacarídeo|alripo|asista|iolhos|odalho|odeme|ramasso|uanabarense|guanabarino)
* heréu
* i(?:migo|mpida|ngrês|nté)
* j(?:au|avaril|ericocim|ugadar|órnea)
* l(?:eixar|erpe|oar|oeste|ucivelo|ucivéu|umiar|üa)
* Lixboa
* m(?:armeluta|eadade|ecê|olher|oínho|uitieramá|uxica)
* n(?:ace|am|aom|aqueste|osomania)
* o(?:bfirmar|bseqüente|cisão|nte|urinol)
* p(?:aje|ascigo|assal|harmácia|ortuguez|roditor)
* querena
* r(?:eferta|oldar|uidento)
* s(?:ainho|alhar|alvanor|apanga|apaão|aturno|ene|ilvano|oer|upresso|uso|énio)
* t(?:artamelar|aubateense|avolagem|eed.+|ábula|ávola)
* unun-héxio
* v(?:ascongadas|assuncê|iado|itando|osco|osmecê|ossancê)

@marcoagpinto One major issue with this rule seems to be that there's no replacement offered. Wouldn't it be better to use the SimpleReplaceRule here, which takes `old_word=replacement_word` in a .txt file?

**Comment by marcoagpinto** [comment]
@danielnaber 
In pt-PT grammar.xml there is no list of words regarding it, so you did well in removing it from the main xml :-)

How would that replacement rule in a .txt file work?

**Comment by danielnaber** [comment]
> How would that replacement rule in a .txt file work?

Just like `barbarisms-pt.txt`.

**Comment by marcoagpinto** [comment]
Ahhhh... seems like a great idea :-)

**Comment by danielnaber** [comment]
Please create `pt/archaisms.txt`, `pt-PT/archaisms-pt-PT.txt` and `pt-BR/archaisms-pt-BR.txt` just like `barbarisms-pt.txt`. I will add the Java code once we have those files (and some data in them to test them).

**Comment by marcoagpinto** [comment]
@danielnaber 

Daniel, the barbarisms ones were in the same folder.

Should the archaisms be also together (same folder) as it would be easier to manage?

**Comment by marcoagpinto** [comment]
@danielnaber 

https://github.com/languagetool-org/languagetool/commit/b76815077009caeb90aa253dcbc9223499860671

**Comment by danielnaber** [comment]
@hmslima  @marcoagpinto Could you provide translations (I assume pt-PT and pt-BR are maybe different) and a link for this, which fit this new rule?

https://github.com/languagetool-org/languagetool/blob/master/languagetool-language-modules/pt/src/main/java/org/languagetool/rules/pt/PortugueseArchaismsRule.java#L72-L92

**Comment by marcoagpinto** [comment]
WORD é um arcaísmo.
WORD é um arcaísmo em Português (Portugal).
WORD é um arcaísmo em Português (Brasil).

Maybe the above?

But what about the regular expressions on grammar.xml regarding the archaisms? They are filled with symbols and almost impossible to decipher.

**Comment by danielnaber** [comment]
> But what about the regular expressions on grammar.xml regarding the archaisms? They are filled with symbols and almost impossible to decipher.

See https://github.com/languagetool-org/languagetool/issues/3095#issuecomment-644340139 where I have cleaned that up already.

**Comment by marcoagpinto** [comment]
@danielnaber 

I don't know what to do as I am confused.

Could you explain in detail?

Thanks!

**Comment by danielnaber** [comment]
`d(?:esemparar|esnobre|evisar|iabro|iese|ino|iésis|olor|orosamente|oroso|díesis)`

becomes this in `archaisms-pt.txt`:

```
desemparar=
desnobre=
devisar=
diabro=
diese=
dino=
diésis=
dolor=
dorosamente=
doroso=
ddíesis=
```

But I don't know what's the replacement, i.e. the right side of the `=` and without that, it doesn't make sense.

**Comment by marcoagpinto** [comment]
@danielnaber 

Yes, we don't know the replacements yet, so it is better to only activate for now:
archaisms-pt-BR.txt
archaisms-pt-PT.txt

**Comment by danielnaber** [comment]
To activate it, I need 4 translations and 1 link: https://github.com/languagetool-org/languagetool/blob/master/languagetool-language-modules/pt/src/main/java/org/languagetool/rules/pt/PortugueseArchaismsRule.java#L72-L92

**Comment by marcoagpinto** [comment]
@danielnaber 

return "Palavras arcaicas evitáveis";
return "Arcaísmo";
return " ou ";
return Tools.getUrl("https://pt.wikipedia.org/wiki/Arcaísmo");

**Comment by marcoagpinto** [comment]
Sorry... I have just edited the above comment as I had forgotten one string.

**Comment by danielnaber** [comment]
> Yes, we don't know the replacements yet, so it is better to only activate for now:
archaisms-pt-BR.txt
archaisms-pt-PT.txt

Done.

**Comment by marcoagpinto** [comment]
Cool. :-)

**Comment by hmslima** [comment]
> Would you be interested in going over all the expressions that are supposed to be archaic to check them? They are listed as regular expressions, which makes it a bit hard to read:

Every single one of them seem to be archaic to me (I didn't know many of them). So, this list seems to be correct.

**Comment by hmslima** [comment]
> Would you be interested in going over all the expressions that are supposed to be archaic to check them? They are listed as regular expressions, which makes it a bit hard to read:

Most words are archaic, except for "[Saturno](https://pt.wikipedia.org/wiki/Saturno_(planeta))". And LanguageTool thinks that "Saturno is a Spanish word, what is correct, but it is a Portuguese word too!

<hr>

But I noticed some _supposed_ mistakes in this list.

> b(?:**abá**|andir|anguê|enção|oa-venturança|odalha|ofé|ródio|uliço)

LT (LanguageTool) already classify the word "babá" as `child word`, so I don't know if this word is useful here. **Apropos**, LT suggests the very same word as replacement... I don't know if this is a bug.

> d(?:esemparar|esnobre|evisar|iabro|iese|ino|iésis|olor|orosamente|oroso|**díesis**)

"ddíesis" has an extra "d".
https://pt.wiktionary.org/wiki/d%C3%ADesis

> g(?:alactooligossacarídeo|alripo|asista|iolhos|odalho|odeme|ramasso|uanabarense|guanabarino)

"guanabarino" has an extra "g".
https://pt.wiktionary.org/wiki/guanabarino

**Comment by marcoagpinto** [comment]
I will take a look tomorrow after I upgrade my Windows 10 to 20.04 :-)

**Comment by marcoagpinto** [comment]
I hope it won't make my computer blow up.

**Comment by marcoagpinto** [comment]
Fixed: https://github.com/languagetool-org/languagetool/commit/fca4f7f08f2f76e49bbb87aa3cc3bf2b20bca3e3

Thanks for the suggestions!

**Comment by marcoagpinto** [comment]
@hmslima 

Hello!

You "câmera" report produced tons of hits in pt-BR:
https://internal1.languagetool.org/regression-tests//20200617/result_pt-BR_20200617_table.html

Are you sure it is Brazilian standard?

**Comment by hmslima** [comment]
@marcoagpinto , Brazilian Portuguese has both "câmera" and "câmara". Here are some examples:

**câmera**
* https://www.americanas.com.br/categoria/cameras-e-filmadoras
* https://www.magazineluiza.com.br/cameras-e-drones/l/cf/
* http://g1.globo.com/pop-arte/cinema/noticia/2010/04/james-cameron-ajuda-desenvolver-camera-3d-para-missao-espacial.html

**câmara**
* https://www.camara.leg.br/
* https://www.amazon.com.br/Harry-Potter-C%C3%A2mara-Secreta-Rowling/dp/8532530796

**Comment by marcoagpinto** [comment]
Ahhhh...

I have just removed the false positives from the file:
https://github.com/languagetool-org/languagetool/commit/3d81caf2bff0e89169addfc84e6ac67fd5c8bc3a
